packages/trader/src/AppV2/Containers/Trade/trade.tsx

```diff
@@ -1,13 +1,191 @@
+        const saved_other_trade_types = localStorage.getItem('other_trade_types');
+        const saved_pinned_trade_types = localStorage.getItem('pinned_trade_types');
+
+        if (saved_other_trade_types) {
+            setOtherTradeTypes(JSON.parse(saved_other_trade_types));
+        }
+
+        if (saved_pinned_trade_types) {
+            setPinnedTradeTypes(JSON.parse(saved_pinned_trade_types));
+        }
+
+        setOpen(true);
+    };
+
+    const handleAddPinnedClick = (item: Item) => {
+        setOtherTradeTypes(prev_categories =>
+            prev_categories.map(category => ({
+                ...category,
+                items: category.items.filter(i => i.id !== item.id).sort((a, b) => a.title.localeCompare(b.title)),
+            }))
+        );
+        setPinnedTradeTypes(prev_pinned => {
+            const updated_pinned = [...prev_pinned];
+            const pinned_category = updated_pinned.find(cat => cat.id === 'pinned');
+
+            if (pinned_category) {
+                pinned_category.items.push(item);
+            } else {
+                updated_pinned.push({
+                    id: 'pinned',
+                    title: 'Pinned',
+                    items: [item],
+                });
+            }
+
+            return updated_pinned;
+        });
+    };
+
+    const handleRemovePinnedClick = (item: Item) => {
+        setPinnedTradeTypes(prev_categories =>
+            prev_categories.map(category => ({
+                ...category,
+                items: category.items.filter(i => i.id !== item.id),
+            }))
+        );
+        setOtherTradeTypes(prev_others => {
+            const updated_others = [...prev_others];
+            const other_category = updated_others.find(cat => cat.id === 'other');
+
+            if (other_category) {
+                other_category.items.unshift(item);
+            } else {
+                updated_others.push({
+                    id: 'other',
+                    items: [item],
+                });
+            }
+
+            updated_others.map(category => ({
+                ...category,
+                items: category.items.sort((a, b) => a.title.localeCompare(b.title)),
+            }));
+
+            return updated_others;
+        });
+    };
+
+    React.useEffect(() => {
+        const formatted_items = createArrayFromCategories(contract_types_list);
+
+        setOtherTradeTypes([
+            {
+                id: 'other',
+                items: formatted_items
+                    .filter(item => !pinned_trade_types[0]?.items.some(pinned_item => pinned_item.id === item.id))
+                    .sort((a, b) => a.title.localeCompare(b.title)),
+            },
+        ]);
+    }, [contract_types_list]);
+
+    React.useEffect(() => {
+        const saved_other_trade_types = localStorage.getItem('other_trade_types');
+        const saved_pinned_trade_types = localStorage.getItem('pinned_trade_types');
+
+        prepareTradeStore();
```

Comment:
```suggestion
        onMount();
```
and 
```
return onUnmount;
```
in useEffect's cleanup :) both methods are from trade-store